EPRS 7.12: Added section for modifying encryption on existing EPRS (#6801)
EnterpriseDB/docs: product_docs/docs/eprs/7/08_xdb_cli/03_xdb_cli_commands/04_encrypt_password.mdx

    - First, obtain the IDs of all users from the Replication Server control database and note down the `sub_db_id` values:

      ```
      SELECT sub_db_id, db_host, db_port, db_user, db_name FROM _edb_replicator_sub.xdb_sub_database;
      ```

    - Then, generate encrypted passwords for each of the Subscription Server users using the Replication Server CLI with `-encrypt` option as previously explained.

    - Finally, update each of the Subscription Server user passwords. Replace `<NEW_ENCRYPTED_PASSWORD>`with the new password and `<SUB_DB_ID>` with the ID previously noted.    

      ```
      UPDATE _edb_replicator_sub.xdb_sub_database SET db_password = '<NEW_ENCRYPTED_PASSWORD>' WHERE sub_db_id = <SUB_DB_ID>;
      ```

1.  Restart the Publication and Subscription Servers and verify that the replication is running successfully. See [Registering a publication server](../../05_smr_operation/02_creating_publication/01_registering_publication_server/#registering_publication_server) to learn how to restart Publication Server.

1.  If you are using the GUI Console, unregister and register all Publication and Subscription servers.

>     If you are using the CLI, create an encrypted password for the admin user and paste it into the Replication Server configuration file: 

Author: @zahid75 did I understand correclty that if one uses the GUI, there is no need to update the config file? Will re-registering Publication and Subscription servers in the GUI take care of updating the password? 🤔

Reviewer: Correct, for GUI an explicit password encryption/configuration is not required.

    ```shell
    #xDB Replication Server Configuration Properties
    admin_password=BThrXGs/gVmStAV88Yl/U2UjoKXVs9qq56j1n/ZnR05Eyg48
    admin_user=admin
    ```

### Bouncy Castle integration 

> Replication Server is also certified with the Bouncy Castle FIPS library. Since the library is not bundled with the Java security framework, you must configure it for use with Replication Server:

Author: @zahid75  I guess the usage of bouncy castle is limited to AES-SHA256 encryption as it is the only one supporting FIPS? And in this case, the library would need to be added before any encryption workflows are performed?

```suggestion
Replication Server is also certified with the Bouncy Castle FIPS library. Since the library is not bundled with the Java security framework, you must configure it for use with Replication Server before using AES-SHA256 to encrypt any passwords :
```

Or is it an either AES-SHA256 or Bouncy Castle?

Reviewer: For first point, your understanding is correct (the Bouncy Castle applies only to AES-SHA256 and not needed for MD5). However the AES-SHA256 is also supported by other providers e.g. SunJCE, and I also tested against it. However SunJCE is not fully compliant with FIPS, so if someone disables use of SunJCE and only want to use Bouncy Castle FIPS provider, then the user must configure BC FIPS jar with EPRS, otherwise it is optional.

To change the encryption method of an existing Replication Server CLI user password, you must create a new password with the new encryption method, and then replace the password on existing objects.

This example changes the encryption algorithm from `DES-MD5` to `AES-SHA256`.

> 1.  Stop the Replication Server Publication and Subscription servers.

Author: ```suggestion
1.  Stop the EPRS Publication and Subscription servers.
```

> 1.  Enable AES-SHA256 encryption in both `xdb_pubserver.conf` and `xdb_subserver.conf` files by uncommenting the `passwordEncryptionMethod=AES-SHA256` line.

Author: ```suggestion
1.  Enable AES-SHA256 encryption in both `xdb_pubserver.conf` and `xdb_subserver.conf` files by uncommenting the `passwordEncryptionMethod=DES-MD5` line and modifying it to `passwordEncryptionMethod=AES-SHA256`.
```

    Create a text file with your password using the editor of your choice. Here, `infile` contains the word `password`.

    ```
    password
    ```

    Execute the `encrypt` command to produce a file named `pwdfile`.

    ```
    $ java -jar edb-repcli.jar -encrypt -input ~/infile -output ~/pwdfile -passwordEncryptionMethod AES-SHA256
    ```

1.  Update the previous DES-MD5 password with the new AES-SHA256 password in the `edb-repl.conf` control file which contains connectivity credentials. You must change the password in two places:

    - Replace the value for `admin_password`, which corresponds to the Replication Server `admin_user`.
    
    - Replace the value for the password that corresponds to the Replication Server control database user.

1.  Create new encrypted passwords for Publication Server users, and replace the previous DES-MD5 passwords with the new AES-SHA256 passwords in the Publication database:

    - First, obtain the IDs of all users from the Replication Server control database and note down the `pub_db_id` values:

      ```
      SELECT pub_db_id, db_host, db_port, db_user, db_name FROM _edb_replicator_pub.xdb_pub_database;
      ```

>     - Then, generate encrypted passwords for each of the Publication Server users using the Replication Server CLI with `-encrypt` option as previously explained.

Reviewer: Replace "Publication Server users" with "Publication database users"

Author: ```suggestion
    - Then, generate encrypted passwords for each of the Publication database users using the Replication Server CLI with `-encrypt` option as previously explained.
```

>     - Finally, update each of the Publication Server user passwords. Replace `<NEW_ENCRYPTED_PASSWORD>` with the new password and `<PUB_DB_ID>` with each of the IDs previously noted.

Reviewer: Replace "Publication Server user" with "Publication database user"

Author: ```suggestion
    - Finally, update each of the Publication database user passwords. Replace `<NEW_ENCRYPTED_PASSWORD>` with the new password and `<PUB_DB_ID>` with each of the IDs previously noted.
```

      ```
      UPDATE _edb_replicator_pub.xdb_pub_database SET db_password = '<NEW_ENCRYPTED_PASSWORD>' WHERE pub_db_id = <PUB_DB_ID>;
      ```

> 1.  Create new encrypted passwords for Subscription Server users, and replace the previous DES-MD5 password with the new AES-SHA256 password in the Subscription database: 

Reviewer: Replace "Subscription Server users" with "Subscription database users"

>     - First, obtain the IDs of all users from the Replication Server control database and note down the `sub_db_id` values:

Author: ```suggestion
    - First, retrieve the IDs of all database connections from the Replication Server control database and note down the `sub_db_id` values:
```

      ```
      SELECT sub_db_id, db_host, db_port, db_user, db_name FROM _edb_replicator_sub.xdb_sub_database;
      ```

>     - Then, generate encrypted passwords for each of the Subscription Server users using the Replication Server CLI with `-encrypt` option as previously explained.

Reviewer: Replace "Subscription Server users" with "Subscription database users"

Author: ```suggestion
    - Then, generate encrypted passwords for each of the Subscription database users using the Replication Server CLI with `-encrypt` option as previously explained.
```

>     - Finally, update each of the Subscription Server user passwords. Replace `<NEW_ENCRYPTED_PASSWORD>`with the new password and `<SUB_DB_ID>` with the ID previously noted.    

Reviewer: Replace "Subscription Server user" with "Subscription database user"

Author: ```suggestion
    - Finally, update each of the Subscription database user passwords. Replace `<NEW_ENCRYPTED_PASSWORD>`with the new password and `<SUB_DB_ID>` with the ID previously noted.    
```

      ```
      UPDATE _edb_replicator_sub.xdb_sub_database SET db_password = '<NEW_ENCRYPTED_PASSWORD>' WHERE sub_db_id = <SUB_DB_ID>;
      ```

1.  Restart the Publication and Subscription Servers and verify that the replication is running successfully. See [Registering a publication server](../../05_smr_operation/02_creating_publication/01_registering_publication_server/#registering_publication_server) to learn how to restart Publication Server.

1.  If you are using the GUI Console, unregister and register all Publication and Subscription servers.

>     If you are using the CLI, create an encrypted password for the admin user and paste it into the Replication Server configuration file: 

Reviewer: Replace "create an encrypted password for the admin user" with "create an encrypted password for the admin user based on AES-SHA256 encryption method".

Author: ```suggestion
    If you are using the CLI, create an encrypted password for the admin user based on the AES-SHA256 encryption method and paste it into the Replication Server configuration file: 
```
